frontend/packages/dev-console/src/components/catalog/catalog-view/CatalogView.tsx

```diff
@@ -0,0 +1,312 @@
+} from '../utils/types';
+
+import CatalogFilters from './CatalogFilters';
+import CatalogToolbar from './CatalogToolbar';
+import CatalogGrid from './CatalogGrid';
+import CatalogCategories from './CatalogCategories';
+import CatalogEmptyState from './CatalogEmptyState';
+import CatalogTypeSelector from './CatalogTypeSelector';
+
+type CatalogViewProps = {
+  items: CatalogItem[];
+  catalogType: string;
+  catalogTypes: CatalogType[];
+  availableCategories: CategoriesType;
+  availableFilters: FiltersType;
+  filterGroups: string[];
+  filterGroupNameMap: CatalogStringMap;
+  filterStoreKey: string;
+  filterRetentionPreference: string[];
+  groupings: CatalogStringMap;
+  renderTile: (item: CatalogItem) => React.ReactNode;
+};
+
+const CatalogView: React.FC<CatalogViewProps> = ({
+  items,
+  catalogType,
+  catalogTypes,
+  availableCategories,
+  availableFilters,
+  filterGroups,
+  filterGroupNameMap,
+  filterStoreKey,
+  filterRetentionPreference,
+  groupings,
+  renderTile,
+}) => {
+  const [activeCategoryId, setActiveCategoryId] = React.useState<string>('all');
+  const [activeSearchKeyword, setActiveSearchKeyword] = React.useState<string>('');
+  const [activeFilters, setActiveFilters] = React.useState<FiltersType>();
+  const [activeGrouping, setActiveGrouping] = React.useState<string>(NoGrouping);
+  const [filterGroupsShowAll, setFilterGroupsShowAll] = React.useState({});
+  const [sortOrder, setSortOrder] = React.useState<CatalogSortOrder>(CatalogSortOrder.ASC);
+  const [filterGroupCounts, setFilterGroupCounts] = React.useState(null);
+
+  const isGrouped = activeGrouping !== NoGrouping;
+
+  const searchInputRef = React.useRef<HTMLInputElement>();
+
+  const itemsSorter = React.useCallback(
+    (itemsToSort) => _.orderBy(itemsToSort, ({ name }) => name.toLowerCase(), [sortOrder]),
+    [sortOrder],
+  );
+
+  const storeFilters = React.useCallback(
+    (filters) => {
+      if (filterStoreKey && filterRetentionPreference) {
+        const filtersToStore = {};
+        _.each(filterRetentionPreference, (filterGroup) => {
+          if (filters[filterGroup]) {
+            filtersToStore[filterGroup] = filters[filterGroup];
+          }
+        });
+        localStorage.setItem(filterStoreKey, JSON.stringify(filtersToStore));
+      }
+    },
+    [filterRetentionPreference, filterStoreKey],
+  );
+
+  const clearFilters = React.useCallback(() => {
+    const params = new URLSearchParams();
+    catalogType && items.length > 0 && params.set('catalogType', catalogType);
+    setURLParams(params);
+
+    const clearedFilters = clearActiveFilters(activeFilters, filterGroups);
+
+    setActiveSearchKeyword('');
+    setActiveFilters(clearedFilters);
+
+    // Don't take focus if a modal was opened while the page was loading.
+    if (!isModalOpen()) {
+      // this doesn't work right now because of issue with PF SearchInput
+      searchInputRef.current && searchInputRef.current.focus({ preventScroll: true });
+    }
```

Comment:
Why is this needed in this callback. Seems unnecessary.

Reply:
This would focus the search input when user clicks clear filters.